src/manifest/dbtProject.ts

```diff
@@ -958,7 +959,54 @@ select * from renamed
     }
     this.queryResultPanel.getSummary(compiledSql, this.adapterType);
   }
+  private getLimitQuery(queryTemplate: string, query: string, limit: number) {
+    return queryTemplate
+      .replace("{query}", () => query)
+      .replace("{limit}", () => limit.toString());
+  }
+  private async getQuery(
+    query: string,
+    limit: number,
+  ): Promise<{ queryTemplate: string; limitQuery: string }> {
+    const queryTemplate = workspace
+      .getConfiguration("dbt")
+      .get<string>("queryTemplate");
+
+    if (queryTemplate && queryTemplate !== DefaultQueryTemplate) {
+      console.log("Using user provided query template", queryTemplate);
+      const limitQuery = this.getLimitQuery(queryTemplate, query, limit);
+
+      return { queryTemplate, limitQuery };
+    }
+
+    try {
+      const dbtVersion = await this.getDBTVersion();
```

Comment:
I think this can be cached. We shouldn't call getDbtVersion() on every query execution. We just need to reevaluate getDbtVersion() every time the python environment is changed.

Reply:
should that be part of this PR?

Reply:
Let's not tackle this as part of this PR.